source/extensions/filters/http/jwt_authn/matcher.h

```diff
@@ -0,0 +1,48 @@
+
+#pragma once
+
+#include "envoy/http/header_map.h"
+
+#include "extensions/filters/http/jwt_authn/verifier.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace JwtAuthn {
+
+class Matcher;
+typedef std::shared_ptr<const Matcher> MatcherConstSharedPtr;
+
+/**
+ *  Wraps RequirementRule configuration, which match request path based on values in
+ *  said configuration and returns matching Verifier.
+ */
+
+class Matcher {
+public:
+  virtual ~Matcher() {}
+
+  // Returns true if path matches requirement configuration, false otherwise.
+  virtual bool matches(const Http::HeaderString& path) const PURE;
+  // Returns the JwtVerifier pointer, or nullptr if not set.
+  virtual const VerifierSharedConstPtr getVerifier() const PURE;
+  // If VerifierSharedConstPtr in this matcher is AllowMissingOrFailedVerifier.
+  virtual bool isAllowMissingOrFailedVerifier() const PURE;
+
+  static MatcherConstSharedPtr createPrefixMatcher(
```

Comment:
can we have one static function to create a matcher?   hide of the logic of creating different matcher inside the function.